components/benefit_sponsors/app/controllers/benefit_sponsors/profiles/registrations_controller.rb

```diff
@@ -44,25 +46,31 @@ def create
                               end
           if is_broker_profile? && saved
             flash[:notice] = "Your registration has been submitted. A response will be sent to the email address you provided once your application is reviewed."
-            render template_filename, :layout => 'single_column'
+            respond_to do |format|
+              format.html { render template_filename, :layout => 'single_column' }
+            end
             return
           elsif saved
             result_url = self.send(result_url)
             redirect_to result_url
             return
           end
-        rescue Exception => e
+        rescue StandardError => e
           flash[:error] = e.message
         end
         params[:profile_type] = profile_type
-        render 'new', :flash => { :error => @agency.errors.full_messages }
+        respond_to do |format|
+          format.html { render 'new', :flash => { :error => @agency.errors.full_messages } }
+        end
       end
 
       def edit
         @agency = BenefitSponsors::Organizations::OrganizationForms::RegistrationForm.for_edit(profile_id: params[:id])
         authorize @agency
 
-        render layout: 'single_column' if @agency.organization.is_broker_profile? || @agency.organization.is_general_agency_profile?
+        respond_to do |format|
```

Comment:
- Request type is already handled in route config **resource_not_found**, this is not needed here. 
- can you revert this logic?